idf_component_manager/cli.py

```diff
@@ -116,7 +116,10 @@ def remove_managed_components(manager):
     manager.remove_managed_components()
 
 
-MANIFEST_COMPONENT_NAME_OPTION = [click.option('--component', default='main', help='Component name in the project')]
+MANIFEST_OPTIONS = [
+    click.option('--component', default='main', help='Component name in the project'),
```

Comment:
These two options should be mutually exclusive. Otherwise, `component` is silently ignored when the path is specified. 

It's also worth mentioning that only components in the `components` directory and `main` are handled.

Reply:
Gotcha. I could pull the component name from path and just ignore component. I could update help to reflect this and raise an error if both are provided.

Reply:
I updated the help messages to be more explicit and raised an error if component and path are provided. I also updated the add dependency function to parse the component name from path when provided.